charmcraft/commands/store/store.py

```diff
@@ -175,7 +175,9 @@ def error_decorator(self, *args, **kwargs):
 class Store:
     """The main interface to the Store's API."""
 
-    def __init__(self, charmhub_config, ephemeral=False):
+    def __init__(self, charmhub_config, ephemeral=False, needs_auth=True):
+        if not needs_auth and not os.getenv(ALTERNATE_AUTH_ENV_VAR):
+            os.environ[ALTERNATE_AUTH_ENV_VAR] = "Tk9LRVk="
```

Comment:
What is this `"Tk9LRVk="` string? Is the environment variable needed? I don't think we should be polluting the environment like this

Reply:
`Tk9LRVk=` is `NOKEY` in base64. There is no way to bypass the key check since it is in `craft_store.StoreClient` `__init__`. It is not known what command will be invoked until it is initialized and return a client for others to call.

Rewrite `craft_store.StoreClient` and `BaseClient` are possible. Move the Auth part out of the `__init__` and check that for each calls. However this requires modifying all functions and cannot be checked at initialization time.

Reply:
Right, I took a look at the code and I see what you're saying. I still think we should find some middle ground so I'm setting up some time where we can discuss this
Thanks!